libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt

```diff
@@ -112,6 +115,8 @@ class RouteProgress internal constructor(
         result = 31 * result + remainingWaypoints
         result = 31 * result + upcomingRoadObjects.hashCode()
         result = 31 * result + stale.hashCode()
+        result = 31 * result + (routeAlternativeId?.hashCode() ?: 0)
+        result = 31 * result + (currentRouteGeometryIndex ?: 0)
```

Comment:
maybe get hash code using `kotlin.HashCodeKt#hashCode` instead of using the value 
```suggestion
        result = 31 * result + currentRouteGeometryIndex.hashCode()
```